assets/src/scripts/components/PageHistory/PageHistory.jsx

```diff
@@ -0,0 +1,181 @@
+import React from 'react';
+import PageHistoryDiff from './PageHistoryDiff.jsx';
+import PageHistoryList from './PageHistoryList.jsx';
+import 'whatwg-fetch';
+
+export default class PageHistory extends React.Component {
+
+	constructor( props ) {
+
+	    super( props );
+
+		this.state = {
+			revisions: this.props.revisions,
+			diff:      { a: null, b: null },
+			page:      1,
+			hasMore:   true,
+			loading:   false,
+		};
+
+	}
+
+	render() {
+
+		var actions = {
+			onSelectRevision: ( revision ) => { this.onSelectRevision( revision ) },
+			onFetchRevisions: () => { this.onfetchRevisions() },
+			onClearDiff:      () => { this.onClearDiff() },
+		};
+
+		return (
+			<div>
+				<PageHistoryDiff diff_a={ this.state.diff.a } diff_b={ this.state.diff.b } />
+				<PageHistoryList revisions={ this.state.revisions } loading={ this.state.loading } hasMore={ this.state.hasMore } actions={ actions } />
+			</div>
+		);
+	}
+
+	onfetchRevisions() {
+
+		if ( ! this.state.hasMore || this.state.loading ) {
+			return;
+		}
+
+		this.setState( { loading: true } );
+
+		var id     = window.HMHandbookPageHistory.post_id;
+		var base   = window.HMHandbookPageHistory.api_base;
+		var nonce  = window.HMHandbookPageHistory.api_nonce;
```

Comment:
Moved all these global settings to a settings file (along with the strings).

Post_ID still comes from the global, but is passed into the component when created.